mu/views/chrome.py

```diff
@@ -0,0 +1,213 @@
+from PyQt5.QtCore import QSize, Qt, pyqtSignal
+from PyQt5.QtWidgets import (
+    QToolBar,
+    QAction,
+    QStackedWidget,
+    QSplashScreen,
+    QDesktopWidget,
+    QWidget,
+    QVBoxLayout,
+    QShortcut,
+    QSplitter,
+    QTabWidget,
+    QFileDialog,
+)
+from PyQt5.QtGui import QKeySequence
+
+from mu.resources import load_icon, load_pixmap
+from mu.views.editor_pane import EditorPane
+
+
+class ButtonBar(QToolBar):
+    """
+    Represents the bar of buttons across the top of the editor and defines
+    their behaviour.
+    """
+
+    slots = None
+    
+    def addAction(self, name, tool_text):
+        action = QAction(
+            load_icon(name),
+            name.capitalize(), self,
+            statusTip=tool_text
+        )
+        super().addAction(action)
+        self.slots[name] = action
+
+    def connect(self, name, slot, *shortcuts):
+        self.slots[name].pyqtConfigure(triggered=slot)
+
+        for shortcut in shortcuts:
+            QShortcut(
+                QKeySequence(shortcut),
+                self.parentWidget()
+            ).activated.connect(slot)
+    
+    def __init__(self, parent):
+        super().__init__(parent)
+
+        self.slots = {}
+        
+        self.setMovable(False)
+        self.setIconSize(QSize(64, 64))
+        self.setToolButtonStyle(3)
+        self.setContextMenuPolicy(Qt.PreventContextMenu)
+        self.setObjectName("StandardToolBar")
+
+        self.addAction(name="new", tool_text="Create a new MicroPython script.")
+        self.addAction(name="load", tool_text="Load a MicroPython script.")
+        self.addAction(
+            name="save",
+            tool_text="Save the current MicroPython script.",
+        )
+        self.addSeparator()
+        self.addAction(
+            name="snippets",
+            tool_text="Use code snippets to help you program.",
+        )
+        self.addAction(
+            name="flash",
+            tool_text="Flash your MicroPython script onto the micro:bit.",
+        )
+        self.addAction(
+            name="repl",
+            tool_text=
+            "Connect to the MicroPython REPL for live coding of the micro:bit.",
+        )
+        self.addSeparator()
+        self.addAction(
+            name="zoom-in",
+            tool_text="Zoom in (to make the text bigger).",
+        )
+        self.addAction(
+            name="zoom-out",
+            tool_text="Zoom out (to make the text smaller).",
+        )
+        self.addSeparator()
+        self.addAction(name="quit", tool_text="Quit the application.")
+
+
+def tab_widget():
```

Comment:
I'd actually prefer to move to using factories for Qt classes, rather than subclassing where possible, but haven't had much time to work on this, hence the scatter-gun approach in getting something mergable and reviewable